daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/XMLTextInfosetOutputter.scala

```diff
@@ -107,15 +160,16 @@ class XMLTextInfosetOutputter private (writer: java.io.Writer, pretty: Boolean,
     outputStartTag(simple)
 
     if (!isNilled(simple) && simple.hasValue) {
-      val text =
-        if (simple.erd.optPrimType.get.isInstanceOf[NodeInfo.String.Kind]) {
-          val s = remapped(simple.dataValueAsString)
-          scala.xml.Utility.escape(s)
+      if (simple.erd.optPrimType.get == NodeInfo.String) {
+        val simpleVal = simple.dataValueAsString
+        if (simple.erd.runtimeProperties.get(XMLTextInfoset.stringAsXml) == "true") {
+          writeStringAsXml(simpleVal)
         } else {
-          simple.dataValueAsString
+          writer.write(scala.xml.Utility.escape(remapped(simpleVal)))
         }
-
-      writer.write(text)
+      } else {
+        writer.write(simple.dataValueAsString)
+      }
     }
 
```

Comment:
`validation="limited"` should work just like normal, as if stringAsXml didn't exist. This is because the "limited" validation we do uses the values on our internal infoset representation.

`validation="on"` will fail, no matter what infoset outputter someone uses. This is because with this mode, internally we `tee` the infoset events to both the InfosetOutputter supplied by the user and the `XMLTextInfosetOutputter`. And since the XMLTextInfosetOutputter results won't match the schema it will always fail.

We could maybe have an option so that the XMLTextInfosetOutputter will disable stringAsXml, so the user gets a stringAsXml-ified infoset, and our Xerces gets a normal infoset with the string escaped. This would allow validation=on to work, but with the caveat that it's only going to validate the stringAsXml stuff as a normal string and not as XML. Should we add that feature? It would just be a boolean flag in the XMLTextInfosetOutputter, but normal users should probably never set it.

Reply:
I would suggest NOT adding that. False sense of security, etc. 

The feature needed is something that edits and creates the modified XSD from the DFDL automatically. I would suggest this isn't Daffodil doing it but some tool. Out of scope for now.